File: StudentEmailGeneratorPython/student_manager.py
class StudentManager:
    
    def __init__(self):
        self.students = {}
    
    def add_student(self, name, student_id, email):
        self.students[student_id] = {
            'name': name,
            'email': email
        }
    
    def get_student_email(self, student_id):
        if student_id in self.students:
            return self.students[student_id]['email']
        return None
    
    def get_all_emails(self):
        return {sid: data['email'] for sid, data in self.students.items()}
    
    def student_exists(self, student_id):
        return student_id in self.students
    
    def get_student_count(self):
        return len(self.students)
    
    def display_all_students(self):
        if not self.students:
            print("No students in the system yet.")
            return
        
        print("\n" + "="*60)
        print("ALL REGISTERED STUDENTS")
        print("="*60)
        for student_id, data in self.students.items():
            print(f"ID: {student_id}")
            print(f"Name: {data['name']}")
            print(f"Email: {data['email']}")
            print("-"*60)

File: StudentEmailGeneratorJava/StudentManager.java
import java.util.HashMap;
import java.util.Map;

public class StudentManager {
    private Map<String, Student> students;
    
    public StudentManager() {
        this.students = new HashMap<>();
    }
    
    public void addStudent(String name, String studentId, String email) {
        students.put(studentId, new Student(name, email));
    }
    
    public String getStudentEmail(String studentId) {
        Student student = students.get(studentId);
        return student != null ? student.getEmail() : null;
    }
    
    public Map<String, String> getAllEmails() {
        Map<String, String> emails = new HashMap<>();
        for (Map.Entry<String, Student> entry : students.entrySet()) {
            emails.put(entry.getKey(), entry.getValue().getEmail());
        }
        return emails;
    }
    
    public boolean studentExists(String studentId) {
        return students.containsKey(studentId);
    }
    
    public int getStudentCount() {
        return students.size();
    }
    
    public void displayAllStudents() {
        if (students.isEmpty()) {
            System.out.println("No students in the system yet.");
            return;
        }
        
        System.out.println("\n" + "=".repeat(60));
        System.out.println("ALL REGISTERED STUDENTS");
        System.out.println("=".repeat(60));
        
        for (Map.Entry<String, Student> entry : students.entrySet()) {
            String studentId = entry.getKey();
            Student student = entry.getValue();
            
            System.out.println("ID: " + studentId);
            System.out.println("Name: " + student.getName());
            System.out.println("Email: " + student.getEmail());
            System.out.println("-".repeat(60));
        }
    }
    
    public Map<String, Student> getStudents() {
        return students;
    }
    
    public void setStudents(Map<String, Student> students) {
        this.students = students;
    }
}

class Student {
    private String name;
    private String email;
    
    public Student(String name, String email) {
        this.name = name;
        this.email = email;
    }
    
    public String getName() {
        return name;
    }
    
    public String getEmail() {
        return email;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public void setEmail(String email) {
        this.email = email;
    }
}
